### PR TITLE
Update postgres parameter check

### LINKED ISSUE #1269: Test connection to the postgres using direct `tokio_postgres::connect` (if getting error details from bb8 is too complex)

### PR DESCRIPTION
**Pull request by Sevenannn**
* Update the error handling for pg_port & pg_host check, previous handling method doesn't give any error messages.
* Use tokio-postgres to test connection using user provided parameters before connecting using bb8 pool. This will immediately propagate any error relevant to user parameter misconfiguration, and therefore avoid the waiting on bb8 connection result and the vague bb8 error message.

### CODE REVIEW

File: crates/db_connection_pool/src/postgrespool.rs [hunk 1]
```diff
@@ -110,8 +117,9 @@ impl PostgresConnectionPool {
                 }
                 if let Some(pg_sslmode) = params.get("pg_sslmode") {
                     match pg_sslmode.to_lowercase().as_str() {
-                        "disable" | "require" | "prefer" | "verify-ca" | "verify-full" => {
```
Comment (ewgenius) [inline crates/db_connection_pool/src/postgrespool.rs:113]: `verify-ca` and `verify-full` not supported by tokio_postgres, but we simulate it by controlling hostname and cert validation, check this lines
https://github.com/spiceai/spiceai/blob/1bdb613457787c938388948b10ac443df89f5526/crates/db_connection_pool/src/postgrespool.rs#L203-L206

File: crates/db_connection_pool/Cargo.toml [hunk 2]
```diff
@@ -34,9 +34,9 @@ odbc-api = { workspace = true, optional = true }
 arrow-odbc = { workspace = true, optional = true }
 lazy_static = "1.4.0"
 clickhouse-rs = { workspace = true, optional = true }
+tokio-postgres = "0.7.10"
```
Comment (ewgenius) [inline crates/db_connection_pool/Cargo.toml]: We alreaady have a dependency for it in workspace

```suggestion
tokio-postgres = { workspace = true }
```

Comment (phillipleblanc) [reply crates/db_connection_pool/Cargo.toml]: This should also be optional and enabled by the `postgres` feature

**Comment by ewgenius** [comment]
closes #1269

File: crates/db_connection_pool/Cargo.toml (hunk 2 above)
Comment (Sevenannn) [reply crates/db_connection_pool/Cargo.toml]: updated to optional tokio postures dependency

**Review by digadeesh — APPROVED** [review]

**Pull request merged**